02-oop-inheritance/Module pattern/script.js

```diff
@@ -0,0 +1,47 @@
+
+var inheritPrototype = function (childClass, parentClass){
+
+    var copyOfParent = Object.create(parentClass.prototype);
+    copyOfParent.constructor = childClass;
+    childClass.prototype = copyOfParent;
+};
+
+var Movie = (function () {
```

Comment:
Yup, I think you're trying to apply the module pattern here. ;)
The idea behind the self invoking function is to wrap all your content in a closure to simulate private data and then return only what you want to make public.
In this case you're confusing your constructor with what would be a wrapper for your entire app, check this:

```
(function () { // This would be your closure.

    // This would be your constructor.
    var Movie = function (rating, title) {
        this.rating = rating;
        this.title = title;
    }

    Movie.prototype.set = function (key, value) {
        this[key] = value;
    }
}());
```

Hope I'm being clear. :)